src/fmt/writer/mod.rs

```diff
@@ -68,7 +115,8 @@ impl Writer {
 ///
 /// The target and style choice can be configured before building.
 pub(crate) struct Builder {
-    target: Target,
+    target_type: TargetType,
+    target_pipe: Option<Arc<Mutex<dyn io::Write + Send + 'static>>>,
```

Comment:
Why split the target into two fields and introduce a new `TargetType` type? I think `target` can just be wrapped in `Option` for the `build` method to work (where you can then use `self.target.take().unwrap_or_default()` to get the actual value).

Reply:
I did try that, because I would like it way more that way, too. But the Target gets passed (and cloned) everywhere (for color checking and the `shim_impl`) that is why `TargetType` needs `Clone, Copy, Debug, Eq, Hash, PartialEq` implemented which is hard to impossible with `dyn io::Write + Send + 'static`.